src/ploigos_step_runner/utils/xml.py

```diff
@@ -87,3 +91,93 @@ def get_xml_element_by_path(xml_file_path, xpath, default_namespace=None, xml_na
         return xml_file.find(xpath)
 
     return xml_file.find(xpath, namespaces)
+
+def aggregate_xml_element_attribute_values(xml_file_path, xml_element, attribs): # pylint: disable=too-many-branches
+    """
+    Function to parse a XML file looking for a specific element and obtaining its attributes.
+
+    If element is not found in file the file is ignored.
+
+    If attribute is not found in element the file is ignored.
+
+    Parameters
+    ----------
+    xml_file_path: str
+        Path to directory containing XML files
+    xml_element: str
+        XML element being searched for
+    attribs: list
+        List of attributes being searched for in the XML element
+
+    Returns
+    --------
+    report_results: dict
+        A dictionary of attribute values.
+
+    Raises
+    ------
+    ValueError
+        If XML element is not found in file(s)
+
+    """
+
+    report_results = {}
+
+    #Search file path for XML files
+
+    if os.path.isdir(xml_file_path):
+        xml_files = glob.glob(xml_file_path + '/*.xml', recursive=False)
+    elif os.path.isfile(xml_file_path):
+        xml_files = [xml_file_path]
+
+    with open(xml_files[0], "r") as xml_file:
+        print("line: " + xml_file.read())
+
+    for attrib in attribs:
+        report_results[attrib] = []
+
+    #Iterate over XML files
+    for file in xml_files:
+        try:
+            pom_element = get_xml_element(file, xml_element)
+            #Add attributes to dictionary
+            for attrib in attribs:
+                attribute_value = pom_element.attrib[attrib]
+                # aggregate attribute accross multiple files
+                report_results[attrib].append(attribute_value)
+        except ValueError:
+            #Return empty dict if an attribute is missing
+            return {}
+
+    #Loop to check if all values aggregated are integers, floats, strings, or a mix
+    #of all three.
+    for attrib in attribs:
+        only_numerics = True
+        only_floats = True
+        #Iteration to determine what pattern all values follow.
+        for value in report_results[attrib]:
+            if only_numerics and value.isnumeric():
```

Comment:
Not a huge thing, but I'd consider making this a nested if statement since once only_numerics is false, it's gonna run line 162 every loop

Reply:
Do you mean putting the try/except in the else?

Reply:
Or an elif that sets only_numerics to false and then the else has the float logic

Reply:
Though if both get set to false there is no point in continuing to iterate through the loop.

Reply:
I was thinking 
`if only_numerics
     if value.isnumeric()
          ...
     else
          only_numerics = false

Reply:
How does this look?

for value in report_results[attrib]:
            if not only_numerics and not only_floats:
                break
            elif only_numerics and value.isnumeric():
                continue
            else:
                only_numerics = False
                try:
                    float(value)
                except ValueError:
                    only_floats = False

Reply:
That didn't format like I wanted lol

Reply:
Maybe this instead:

```for value in report_results[attrib]:
            if not only_numerics and not only_floats:
                break
            if value.isnumeric():
                continue
            only_numerics = False
            try:
                float(value)
            except ValueError:
                only_floats = False```